src/ucs/sys/compiler_def.h

```diff
@@ -92,6 +92,11 @@
 #define UCS_PTR_BYTE_OFFSET(_ptr, _offset) \
     ((void *)((uintptr_t)(_ptr) + (_offset)))
 
+/* Helper macro for address arithmetic in elements of type _type */
```

Comment:
maybe remove `_offset`? It is almost always 1
if need to move to more than 1 element BYTE_OFFSET can be used